### ISSUE TITLE
wrong triangular inverse

### ISSUE THREAD

**Issue by pachadotdev** [pachadotdev/fixest2#23]
There is an error in https://github.com/pachadotdev/fixest2/blob/cpp11-some-r-refactoring/src/04_02_linear_model_cholesky.cpp#L3

See this example, position row 2, col 1 in the matrix "R" gives an incorrect value

the correct value (done in the whiteboard) should be

```
 0.177 -0.591
-0.591  0.184
``` 

Using cpp11 all the time (https://github.com/pachadotdev/fixest2/blob/cpp11-some-r-refactoring/src/04_02_linear_model_cholesky.cpp#L104-L179)
```
devtools::load_all()

X = cbind(1, mtcars$wt)
y = mtcars$mpg
w = 1
colnames(X) = c("(Intercept)","mpg")

correct_0w = F
nthreads = 1
collin.tol = 10^(-5)

info_products = cpp_sparse_products(X, w, y, correct_0w, nthreads)
xwx = info_products$XtX
xwy = info_products$Xty
info_inv = cpp_cholesky(xwx, collin.tol, nthreads)

info_inv$R

              [,1]     [,2]
[1,]  5.656854e+00 18.19951
[2,] 4.651574e-310  5.44782
````

Using Rcpp code for `cpp_cholesky` and therefore use Rcpp for the matrix "R" (https://github.com/pachadotdev/fixest2/blob/cpp11-some-r-refactoring/dev/correct_r_matrix.cpp#L114-L181)
```
devtools::load_all()
Rcpp::sourceCpp("dev/correct_r_matrix.cpp")

X = cbind(1, mtcars$wt)
y = mtcars$mpg
w = 1
colnames(X) = c("(Intercept)","mpg")

correct_0w = F
nthreads = 1
collin.tol = 10^(-5)

info_products = cpp_sparse_products(X, w, y, correct_0w, nthreads)
xwx = info_products$XtX
xwy = info_products$Xty
info_inv = cpp_cholesky(xwx, collin.tol, nthreads)

info_inv$R

           [,1]       [,2]
[1,]  0.1767767 -0.5905573
[2,] -0.5905573  0.1835596
```

Update:

I created a minimal example where I get the same matrices, which makes me suspect that the problem is in these lines https://github.com/pachadotdev/fixest2/blob/cpp11-some-r-refactoring/src/04_02_linear_model_cholesky.cpp#L134

the MWE is https://github.com/pachadotdev/fixest2/blob/cpp11-some-r-refactoring/dev/check_r_matrix_from_cpp_cholesky.R

**Comment by pachadotdev** [comment]
hi @DavisVaughan
happy holidays!
could I please have your help with this issue? I am doing something wrong in expressing the idea within cpp11 grammar

**Comment by pachadotdev** [comment]
fixed in efb8d422c9ea57ea71b851112c02be5522d3500b
not the most efficient solution, but better than many for loops
